webrender/src/platform/macos/font.rs

```diff
@@ -72,76 +100,96 @@ impl FontContext {
-                ct_font
+                Some(ct_font)
             }
-        };
-
-        let glyph = character as CGGlyph;
-        let bounds = ct_font.get_bounding_rects_for_glyphs(kCTFontDefaultOrientation, &[glyph]);
-
-        let rasterized_left = bounds.origin.x.floor() as i32;
-        let rasterized_width =
-            (bounds.origin.x - (rasterized_left as f64) + bounds.size.width).ceil() as u32;
-        let rasterized_descent = (-bounds.origin.y).ceil() as i32;
-        let rasterized_ascent = (bounds.size.height + bounds.origin.y).ceil() as i32;
-        let rasterized_height = (rasterized_descent + rasterized_ascent) as u32;
-        if rasterized_width == 0 || rasterized_height == 0 {
-            return Some(RasterizedGlyph::blank())
         }
+    }
 
-        let mut cg_context = CGContext::create_bitmap_context(rasterized_width as usize,
-                                                              rasterized_height as usize,
-                                                              8,
-                                                              rasterized_width as usize * 4,
-                                                              &CGColorSpace::create_device_rgb(),
-                                                              kCGImageAlphaPremultipliedLast);
-        cg_context.set_allows_font_smoothing(enable_aa);
-        cg_context.set_should_smooth_fonts(enable_aa);
-        cg_context.set_allows_antialiasing(enable_aa);
-        cg_context.set_should_antialias(enable_aa);
-        cg_context.set_rgb_fill_color(1.0, 1.0, 1.0, 1.0);
-
-        let rasterization_origin = CGPoint {
-            x: -rasterized_left as f64,
-            y: rasterized_descent as f64,
-        };
-        ct_font.draw_glyphs(&[glyph], &[rasterization_origin], cg_context.clone());
-
-        let rasterized_area = (rasterized_width * rasterized_height) as usize;
-        let mut rasterized_pixels = cg_context.data().to_vec();
-        for i in 0..rasterized_area {
-            let alpha = (rasterized_pixels[i * 4 + 3] as f32) / 255.0;
-            for j in 0..3 {
-                rasterized_pixels[i * 4 + j] =
-                    ((rasterized_pixels[i * 4 + j] as f32) / alpha) as u8;
+    #[allow(dead_code)]     // TODO(gw): Expose this to the public glyph dimensions API.
+    pub fn get_glyph_dimensions(&mut self,
+                                font_key: FontKey,
+                                size: Au,
+                                character: u32,
+                                device_pixel_ratio: f32) -> Option<(Au, Au)> {
+        self.get_ct_font(font_key, size, device_pixel_ratio).map(|ref ct_font| {
+            let glyph = character as CGGlyph;
+            let metrics = get_glyph_metrics(ct_font, glyph);
+            (Au::from_px(metrics.rasterized_width as i32), Au::from_px(metrics.rasterized_height as i32))
+        })
+    }
+
+    pub fn get_glyph(&mut self,
```

Comment:
Perhaps, `rasterize_glyph`? to reflect that this is a heavy operation (contrary to most of the get_* routines being just accessors)